### PR TITLE
faucet cli improvements

### LINKED ISSUE #336: [cli] clean up faucet cli
The current state is working, but not convenient:

a sub-command named faucet and then drip,create,dissolve,close would be nicer. But we also need to disambiguate with the existing faucet command which requires no reputation.....so for now I can live with this wild jungle of commands. But:

* [x] better explain cli commands: 
   * close-faucet: lazy garbage collection. can only be called by faucet creator and only once the faucet is empty
   * dissolve-faucet: can only be called by root. moreover, explain arguments: first one is the faucet address, second one is the beneficiary of the remainding funds
* [x] improve create-faucet
   * whitelist should be comma-separated to make it clear that it is a single argument. explain in cli help
   * the return value of this command needs to be the faucet address. First, check extrinsic success, report error, then parse FaucetCreated event for address
* [x] drip-faucet: check extrinsic success and extract error if any
* [x] add list-faucets command.List all faucet accounts, their name, balance, drip amount and whitelist

with the upgrade of the api-client, this should all be easier now. (examples are not there yet, unfortunately: https://github.com/scs/substrate-api-client/issues/591)

### PR DESCRIPTION
**Pull request by pifragile**
*Closes #336

### CODE REVIEW

File: client/src/cli_args.rs [hunk 1]
```diff
@@ -172,8 +172,8 @@ impl<'a, 'b> EncointerArgs<'b> for App<'a, 'b> {
 			Arg::with_name(WHITELIST_ARG)
 				.takes_value(true)
 				.required(false)
-				.multiple(true)
-				.min_values(1),
+				.value_name("WHITELIST")
```
Comment (brenzi) [inline client/src/cli_args.rs:175]: I just realize that people may use spaces after the comma which breaks this argument: maybe indicate in help that no spaces may be used?

File: client/src/main.rs [hunk 2]
```diff
@@ -1657,14 +1657,30 @@ async fn main() {
...
+
+                    let result = api.submit_and_watch_extrinsic_until_success(xt, false);
+
+                    match result {
+                        Ok(report) => {
+                            let events = report
+                            .events
+                            .unwrap();
+
+                            println!("{}", AccountId::decode(&mut events[5].field_bytes()[0..32].as_ref()).unwrap().to_ss58check());
```
Comment (brenzi) [inline client/src/main.rs]: thread 'main' panicked at 'range end index 32 out of range for slice of length 28', client/src/main.rs:1670:67
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

Comment (brenzi) [reply client/src/main.rs]: faucet was created, though

Comment (brenzi) [reply client/src/main.rs]: image

**Review by brenzi — CHANGES_REQUESTED** [review]

File: client/src/main.rs (hunk 2 above)
Comment (brenzi) [reply client/src/main.rs]: I think it is bad practice to use hardcoded indexing (especially for the event index). runtime / pallet changes can break your index assumptions and render this very brittle

**Comment by brenzi** [comment]
all the rest LGTM. thx!

File: client/src/main.rs (hunk 2 above)
Comment (brenzi) [reply client/src/main.rs]: you should be able to loop through all events and attempt to decode each into the original EventType RuntimeEvents::FaucetCreated. if successful, extract accountid

Comment (brenzi) [reply client/src/main.rs]: was an easy fix. committed directly

**Review by brenzi — APPROVED** [review]

File: client/src/main.rs (hunk 2 above)
Comment (pifragile) [reply client/src/main.rs]: thanks for fixing!

**Pull request merged**